docs/appconfiguration_generator.md

```diff
@@ -0,0 +1,179 @@
+        trait.NewOpsRuleGeneratorFunc(g.project, g.stack, g.appName, g.app),
+        NewMonitoringGeneratorFunc(g.project, g.app.Monitoring, g.appName),
+        // The OrderedResourcesGenerator should be executed after all resources are generated.
+        NewOrderedResourcesGeneratorFunc(),
+}
+```
+The `WorkloadGenerator` needs to handle tasks such as injecting monitoring-related annotations and labels, which should be the responsibility of the `MonitoringGenerator`.
+```go
+monitoringLabels := make(map[string]string)
+monitoringAnnotations := make(map[string]string)
+if g.monitoring != nil {
+    if g.project.ProjectConfiguration.Prometheus != nil && g.project.ProjectConfiguration.Prometheus.OperatorMode {
+        monitoringLabels["kusion_monitoring_appname"] = g.appName
+    } else if g.project.ProjectConfiguration.Prometheus != nil && !g.project.ProjectConfiguration.Prometheus.OperatorMode {
+        // If Prometheus doesn't run as an operator, kusion will generate the
+        // most widely-known annotation for workloads that can be consumed by
+        // the out-of-the-box community version of Prometheus server
+        // installation shown as below:
+        monitoringAnnotations["prometheus.io/scrape"] = "true"
+        monitoringAnnotations["prometheus.io/scheme"] = g.monitoring.Scheme
+        monitoringAnnotations["prometheus.io/path"] = g.monitoring.Path
+        monitoringAnnotations["prometheus.io/port"] = g.monitoring.Port
+    }
+}
+```
+**Example 2:**
+
+The `DataBaseGenerator` needs to inject env in the corresponding resources of the `WorkloadGenerator`. The current implementation requires the `DataBaseGenerator` to be executed before the `WorkloadGenerator`.
+```go
+func (g *databaseGenerator) injectSecret(secret *v1.Secret) error {
+    // Inject the database information into the containers of service workload.
+    if g.workload.Service != nil {
+        for _, v := range g.workload.Service.Containers {
+            v.Env[dbHostAddressEnv] = "secret://" + secret.Name + "/hostAddress"
+            v.Env[dbUsernameEnv] = "secret://" + secret.Name + "/username"
+            v.Env[dbPasswordEnv] = "secret://" + secret.Name + "/password"
+        }
+    }
+    
+    // Inject the database information into the containers of job workload.
+    if g.workload.Job != nil {
+        for _, v := range g.workload.Job.Containers {
+            v.Env[dbHostAddressEnv] = "secret://" + secret.Name + "/hostAddress"
+            v.Env[dbUsernameEnv] = "secret://" + secret.Name + "/username"
+            v.Env[dbPasswordEnv] = "secret://" + secret.Name + "/password"
+        }
+    }
+    
+    return nil
+}
+```
+## Goal
+1. Refactor the code in the Generator section to improve its extensibility, making it easier and more maintainable to add `Generator`s.
+
+## Non-Goal
+1. It is not considered to split `Generator` out of this code repository and put it into a separate repository such as Generator-Catalog.
```

Comment:
We will move generators into [catalog](https://github.com/KusionStack/catalog) in the future, but this will not affect your current refactor.